apollo-router/src/plugins/external.rs

```diff
@@ -57,227 +188,247 @@ struct BaseConf {
     sdl: bool,
 }
 
+/// What information is passed to a subgraph request/response stage
 #[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize, JsonSchema)]
-#[serde(default)]
-struct RouterStage {
-    /// The request configuration
-    request: Option<BaseConf>,
-    /// The response configuration
-    response: Option<BaseConf>,
+#[serde(default, deny_unknown_fields)]
+struct SubgraphConf {
+    /// Send the headers
+    headers: bool,
+    /// Send the context
+    context: bool,
+    /// Send the body
+    body: bool,
+    /// Send the service name
+    service: bool,
+    /// Send the subgraph URI
+    uri: bool,
+    /// Send the service name
+    service_name: bool,
 }
 
 /// The stages request/response configuration
 #[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize, JsonSchema)]
-#[serde(default)]
+#[serde(default, deny_unknown_fields)]
 struct Stages {
     /// The router stage
-    router: Option<RouterStage>,
+    router: RouterStage,
+    /// The subgraph stage
+    subgraph: SubgraphStage,
 }
 
 /// Configures the externalization plugin
 #[derive(Clone, Debug, Default, Deserialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
 struct Conf {
     /// The url you'd like to offload processing to
     url: String,
     /// The timeout for external requests
-    #[serde(deserialize_with = "humantime_serde::deserialize", default)]
-    #[schemars(with = "String", default)]
-    timeout: Option<Duration>,
+    #[serde(deserialize_with = "humantime_serde::deserialize")]
+    #[schemars(with = "String", default = "default_timeout")]
+    #[serde(default = "default_timeout")]
+    timeout: Duration,
     /// The stages request/response configuration
     #[serde(default)]
-    stages: Option<Stages>,
+    stages: Stages,
 }
 
-#[async_trait::async_trait]
-impl Plugin for ExternalPlugin {
-    type Config = Conf;
+fn default_timeout() -> Duration {
+    DEFAULT_EXTERNALIZATION_TIMEOUT
+}
 
-    async fn new(init: PluginInit<Self::Config>) -> Result<Self, BoxError> {
-        Ok(ExternalPlugin {
-            configuration: init.config,
-            sdl: init.supergraph_sdl,
-        })
-    }
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize, JsonSchema)]
+#[serde(default)]
+struct RouterStage {
+    /// The request configuration
+    request: RouterConf,
+    /// The response configuration
+    response: RouterConf,
+}
```

Comment:
Both of these should be optional in configuration...

So:
```
plugins:
  experimental.external:
    # A URL which will be called for each request for any configured stage.
    url: http://127.0.0.1:8081
    stages:
      router:
        request:
          headers: true
 ```
should still be a valid config. i.e.: specifying something for the router request, but not for router response. It looks like the change has made both request and response mandatory in config.

Reply:
@BrynCooke and I went for a default impl that is disabled (IIRC, @BrynCooke lmk if i got it wrong) so i think we're good here